src/passes/Asyncify.cpp

```diff
@@ -1543,16 +1546,32 @@ struct Asyncify : public Pass {
   }
 
 private:
-  void addGlobals(Module* module) {
+  void addGlobals(Module* module, bool imported) {
     Builder builder(*module);
-    module->addGlobal(builder.makeGlobal(ASYNCIFY_STATE,
-                                         Type::i32,
-                                         builder.makeConst(int32_t(0)),
-                                         Builder::Mutable));
-    module->addGlobal(builder.makeGlobal(ASYNCIFY_DATA,
-                                         Type::i32,
-                                         builder.makeConst(int32_t(0)),
-                                         Builder::Mutable));
+    {
+      auto asyncifyState = builder.makeGlobal(ASYNCIFY_STATE,
+                                               Type::i32,
+                                               builder.makeConst(int32_t(0)),
+                                               Builder::Mutable);
+      if (imported) {
+        asyncifyState->module = ENV;                                         
+        asyncifyState->base = ASYNCIFY_STATE;  
+      }                                         
+
+      module->addGlobal(std::move(asyncifyState)); 
+    }
+    {
+      auto asyncifyData = builder.makeGlobal(ASYNCIFY_DATA,
+                                               Type::i32,
+                                               builder.makeConst(int32_t(0)),
+                                               Builder::Mutable);
+      if (imported) {
+        asyncifyData->module = ENV;                                         
+        asyncifyData->base = ASYNCIFY_DATA;  
+      }                                           
+
+      module->addGlobal(std::move(asyncifyData)); 
+    }
```

Comment:
Why these extra scopes?

Reply:
It was the avoidance of variable name typo. Removed because it is no longer necessary.